spark/tests/test_spark.py

```diff
@@ -894,6 +878,60 @@ def test_standalone_unit(aggregator):
         aggregator.assert_all_metrics_covered()
 
 
+@pytest.mark.unit
+def test_standalone_unit_with_proxy_warning_page(aggregator):
+    c = SparkCheck('spark', {}, [STANDALONE_CONFIG_WITH_SESSION_PROXY])
+    with mock.patch('requests.sessions.Session.get', proxy_with_warning_page_mock):
+        c.check(STANDALONE_CONFIG_WITH_SESSION_PROXY)
+
+        # Check the running job metrics
+        for metric, value in iteritems(SPARK_JOB_RUNNING_METRIC_VALUES):
+            aggregator.assert_metric(metric, value=value, tags=SPARK_JOB_RUNNING_METRIC_TAGS)
+
+        # Check the running job metrics
+        for metric, value in iteritems(SPARK_JOB_RUNNING_METRIC_VALUES):
+            aggregator.assert_metric(metric, value=value, tags=SPARK_JOB_RUNNING_METRIC_TAGS)
+
+        # Check the succeeded job metrics
+        for metric, value in iteritems(SPARK_JOB_SUCCEEDED_METRIC_VALUES):
+            aggregator.assert_metric(metric, value=value, tags=SPARK_JOB_SUCCEEDED_METRIC_TAGS)
+
+        # Check the running stage metrics
+        for metric, value in iteritems(SPARK_STAGE_RUNNING_METRIC_VALUES):
+            aggregator.assert_metric(metric, value=value, tags=SPARK_STAGE_RUNNING_METRIC_TAGS)
+
+        # Check the complete stage metrics
+        for metric, value in iteritems(SPARK_STAGE_COMPLETE_METRIC_VALUES):
+            aggregator.assert_metric(metric, value=value, tags=SPARK_STAGE_COMPLETE_METRIC_TAGS)
+
+        # Check the driver metrics
+        for metric, value in iteritems(SPARK_DRIVER_METRIC_VALUES):
+            aggregator.assert_metric(metric, value=value, tags=SPARK_METRIC_TAGS)
+
+        # Check the executor metrics
+        for metric, value in iteritems(SPARK_EXECUTOR_METRIC_VALUES):
+            aggregator.assert_metric(metric, value=value, tags=SPARK_METRIC_TAGS)
+
+        # Check the RDD metrics
+        for metric, value in iteritems(SPARK_RDD_METRIC_VALUES):
+            aggregator.assert_metric(metric, value=value, tags=SPARK_METRIC_TAGS)
+
+        # Check the streaming statistics metrics
+        for metric, value in iteritems(SPARK_STREAMING_STATISTICS_METRIC_VALUES):
+            aggregator.assert_metric(metric, value=value, tags=SPARK_METRIC_TAGS)
+
+        # Check the service tests
+        for sc in aggregator.service_checks(STANDALONE_SERVICE_CHECK):
+            assert sc.status == SparkCheck.OK
+            assert sc.tags == ['url:http://localhost:8080', 'cluster_name:SparkCluster']
+        for sc in aggregator.service_checks(SPARK_SERVICE_CHECK):
+            assert sc.status == SparkCheck.OK
+            assert sc.tags == ['url:http://localhost:4040', 'cluster_name:SparkCluster']
+
+        # Assert coverage for this check on this instance
+        aggregator.assert_all_metrics_covered()
```

Comment:
Note: I don't think these lines *need* to be within the `mock` context, but… that's what we do in the other tests, so… okay. :-)